modules/library/main/src/main/java/org/geotools/styling/ContrastEnhancementImpl.java

```diff
@@ -51,38 +51,57 @@
  * &lt;xs:element name=&quot;GammaValue&quot; type=&quot;xs:double&quot;/&gt;
  * </pre>
  * 
- * In the case of a color image, the relative grayscale brightness of a pixel color is used.
- * ?Normalize? means to stretch the contrast so that the dimmest color is stretched to black and the
- * brightest color is stretched to white, with all colors in between stretched out linearly.
- * ?Histogram? means to stretch the contrast based on a histogram of how many colors are at each
- * brightness level on input, with the goal of producing equal number of pixels in the image at each
- * brightness level on output. This has the effect of revealing many subtle ground features. A
- * ?GammaValue? tells how much to brighten (value greater than 1.0) or dim (value less than 1.0) an
- * image. The default GammaValue is 1.0 (no change). If none of Normalize, Histogram, or GammaValue
- * are selected in a ContrastEnhancement, then no enhancement is performed.
+ * In the case of a color image, the relative grayscale brightness of a pixel color is used. ?Normalize? means to stretch the contrast so that the
+ * dimmest color is stretched to black and the brightest color is stretched to white, with all colors in between stretched out linearly. ?Histogram?
+ * means to stretch the contrast based on a histogram of how many colors are at each brightness level on input, with the goal of producing equal
+ * number of pixels in the image at each brightness level on output. This has the effect of revealing many subtle ground features. A ?GammaValue?
+ * tells how much to brighten (value greater than 1.0) or dim (value less than 1.0) an image. The default GammaValue is 1.0 (no change). If none of
+ * Normalize, Histogram, or GammaValue are selected in a ContrastEnhancement, then no enhancement is performed.
  * 
  * @author iant
  *
  *
- * @source $URL$
- *         http://svn.osgeo.org/geotools/trunk/modules/library/main/src/main/java/org/geotools/
- *         styling/ContrastEnhancementImpl.java $
+ * @source $URL$ http://svn.osgeo.org/geotools/trunk/modules/library/main/src/main/java/org/geotools/ styling/ContrastEnhancementImpl.java $
  */
 public class ContrastEnhancementImpl implements ContrastEnhancement {
+
     private FilterFactory filterFactory;
 
     private Expression gamma;
 
     private Expression type;
-
+    
     private ContrastMethod method;
 
+    private Map<String, Expression> options;
+
     public ContrastEnhancementImpl() {
         this(CommonFactoryFinder.getFilterFactory(null));
     }
 
     public ContrastEnhancementImpl(FilterFactory factory) {
-        this(factory, null);
+        this(factory, (ContrastMethod)null);
+    }
+
+    /**
+     * @param method
+     * @param filterFactory2 
+     */
+    private void lookupMethod(String method, FilterFactory ff) {
+        if(ff==null) {
+            ff=CommonFactoryFinder.getFilterFactory();
+        }
+        if (ContrastMethod.NORMALIZE.matches(method)) {
```

Comment:
Consider moving this logic to ContrastMethod so the logic does not get repeated elsewhere.